### PR TITLE
Create and fill login controller

### PR DESCRIPTION
**Pull request by grom4er**

### CODE REVIEW

File: src/main/java/taxiservice/web/filters/AuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+        String url = request.getServletPath();
+        if (url.equals("/drivers/create") || url.equals("/drivers/login")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        Long driverId = (Long) request.getSession().getAttribute(USER_ID);
+        if (driverId == null || driverService.get(driverId) == null){
```
Comment (Max-Pochepets) [inline src/main/java/taxiservice/web/filters/AuthenticationFilter.java:40]: paragraph 5 [checklist](https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/09_filters_checklist.html)

File: src/main/webapp/WEB-INF/drivers/create.jsp [hunk 2]
```diff
@@ -9,6 +9,8 @@
 <form method="post" action="${pageContext.request.contextPath}/drivers/create">
     Driver name:<input type="text" name="name" pattern="^([A-Z])[a-z]*">
     Driver license number:<input type="text" name="license">
+    Driver login:<input type="text" name="login" pattern="^a-z0-9]*">
+    Driver password number:<input type="text" name="pwd">
```
Comment (Max-Pochepets) [inline src/main/webapp/WEB-INF/drivers/create.jsp:13]: the password, while typed, could be seen. can we fix it?

Comment (Max-Pochepets) [reply src/main/webapp/WEB-INF/drivers/create.jsp:13]: as well as blank input

File: src/main/webapp/WEB-INF/drivers/login.jsp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+<head>
+    <title>Login</title>
+</head>
+<body>
+<h1> Login page</h1>
+<h4 style ="color:red">${error}</h4>
+
+<form method="post" action="${pageContext.request.contextPath}/drivers/login">
+    Driver login:<input type="text" name="login" pattern="^[a-z]*">
+    Driver password :<input type="text" name="pwd">
```
Comment (Max-Pochepets) [inline src/main/webapp/WEB-INF/drivers/login.jsp:12]: same

**Review by Max-Pochepets — APPROVED** [review]
fix travis :)

File: src/main/java/taxiservice/web/filters/AuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+        String url = request.getServletPath();
+        if (url.equals("/drivers/create") || url.equals("/drivers/login")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        Long driverId = (Long) request.getSession().getAttribute(DRIVER_ID);
+        if (driverService.get(driverId) == null) {
```
Comment (RomanDubovskyi) [inline src/main/java/taxiservice/web/filters/AuthenticationFilter.java:42]: it can't be null, it either exists, or you will get NoSuchElementException

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
don't forget to fix points, I mentioned in the generate chat

File: src/main/java/taxiservice/web/filters/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+        String url = request.getServletPath();
+        if (ALLOWED_URL.contains(url)) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        Long driverId = (Long) request.getSession().getAttribute(DRIVER_ID);
+        if (driverService.get(driverId) == null) {
```
Comment (RomanDubovskyi) [inline src/main/java/taxiservice/web/filters/AuthenticationFilter.java:46]: what about this comment?

File: src/main/java/taxiservice/web/filters/AuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import taxiservice.lib.Injector;
+import taxiservice.service.DriverService;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+    private static final Injector injector =
+            Injector.getInstance(Application.class.getPackageName());
+    private static final DriverService driverService =
+            (DriverService) injector.getInstance(DriverService.class);
+    private static final Set<String> ALLOWED_URL = new HashSet<>();
```
Comment (RomanDubovskyi) [inline src/main/java/taxiservice/web/filters/AuthenticationFilter.java:24]:
```suggestion
    private static final Set<String> allowedUrls = new HashSet<>();
```
it's not a constant (not deeply immutable, the same situation as with injector)

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**